src/ray/raylet/node_manager.cc

```diff
@@ -344,6 +344,68 @@ void NodeManager::ProcessClientMessage(std::shared_ptr<LocalClientConnection> cl
     ObjectID object_id = from_flatbuf(*message->object_id());
     RAY_LOG(DEBUG) << "reconstructing object " << object_id;
     RAY_CHECK_OK(object_manager_.Pull(object_id));
+
+    // If the blocked client is a worker, then release any CPU resources that
+    // it acquired for its assigned task while it is blocked. The resources
+    // will be acquired again once the worker is unblocked.
+    std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
+    if (worker) {
+      // Only release the resources if the worker isn't already blocked.
+      if (!worker->IsBlocked()) {
+        RAY_CHECK(!worker->GetAssignedTaskId().is_nil());
+        auto tasks = local_queues_.RemoveTasks({worker->GetAssignedTaskId()});
+        const auto &task = tasks.front();
+        // Get the CPU resources required by the running task.
+        const auto required_resources =
+            task.GetTaskSpecification().GetRequiredResources();
+        double required_cpus;
```

Comment:
initialize `required_cpus` to zero. Valgrind will legitimately complain when you pass a pointer to uninitialized memory .